service/src/test/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsServiceTest.java

```diff
@@ -0,0 +1,243 @@
+    JacksonAutoConfiguration.class,
+    UploadDocumentsService.class,
+    FixedTimeConfiguration.class
+})
+class UploadDocumentsServiceTest {
+
+    private static final String USER = "HMCTS";
+    private CaseData givenCaseData;
+
+    @Autowired
+    private Time time;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private UploadDocumentsService uploadDocumentsService;
+
+    @MockBean
+    private DocumentUploadHelper documentUploadHelper;
+
+    @BeforeEach
+    void setup() {
+        givenCaseData = prepareCaseData();
+    }
+
+    @Test
+    void shouldReturnMapOfCaseDetailsWithAttachedDocuments() {
+        CaseData caseData = caseData();
+        when(documentUploadHelper.getUploadedDocumentUserDetails()).thenReturn(USER);
+
+        Map<String, Object> map = uploadDocumentsService.updateCaseDocuments(caseData, caseData);
+
+        assertThat(map.get("documents_socialWorkChronology_document"))
+            .isEqualToComparingOnlyGivenFields(Document.builder()
+                .uploadedBy(USER)
+                .dateTimeUploaded(time.now())
+                .build());
+
+        assertThat(map.get("documents_socialWorkStatement_document"))
+            .isEqualToComparingOnlyGivenFields(Document.builder()
+                .statusReason("Social work statement and genogram text")
+                .uploadedBy(USER)
+                .dateTimeUploaded(time.now())
+                .documentStatus("To follow")
+                .build());
+
+        assertThat(map.get("documents_checklist_document"))
+            .isEqualToComparingOnlyGivenFields(Document.builder()
+                .statusReason("Social work chronology text")
+                .uploadedBy(USER)
+                .dateTimeUploaded(time.now())
+                .documentStatus("To follow")
+                .build());
+    }
+
+    @Test
+    void shouldUpdateOtherSocialWorkDocumentsListWithUpdatedDetailsAndUser() {
+        when(documentUploadHelper.getUploadedDocumentUserDetails()).thenReturn(USER);
+
+        List<DocumentSocialWorkOther> list = unwrapElements(
+            uploadDocumentsService.setUpdatedByAndDateAndTimeOnDocuments(
+                createCaseDataWithUpdatedDocumentSocialWorkOther(),
+                createCaseDataWithOldDocumentSocialWorkOther()));
+
+        assertThat(list).first()
+            .extracting(DocumentSocialWorkOther::getTypeOfDocument)
+            .extracting(DocumentReference::getUrl)
+            .isEqualTo("/new_test.doc");
+
+        assertThat(list).first()
+            .extracting(
+                DocumentSocialWorkOther::getDocumentTitle,
+                DocumentSocialWorkOther::getDateTimeUploaded,
+                DocumentSocialWorkOther::getUploadedBy)
+            .containsExactly(
+                "New Additional Doc 1",
+                time.now(),
+                USER);
+    }
+
+    @Test
+    void shouldUpdateOtherSocialWorkDocumentsListWithNewDocument() {
```

Comment:
Do we have a test that checks author is overwritten in overwrite action? And a test that asserts previous uploadedBy and author details are persisted within non overwrite action.

Reply:
it's already included in existing test.